source/language/minimal_profile.md

```diff
@@ -0,0 +1,130 @@
+
+##  Language Requirements
+
+1.   MAY NOT use `include` statements, other than `include "stdgates.inc"`
+	-  Why: I am not convinced that `include` is used much. I would
+       also assume that the search path depends on implementation
+       specific details.
+2. MAY NOT declare or use *variables* of any classical type
+	- Why: Implementations are probably quite uneven both in the
+      types, operations, and circumstances in which classical data
+      computed at run-time may be used. It's not needed for many
+      "textbook" quantum circuits.
+3. MAY use `float` literals
+	- Why: needed for parametric gates
+4. MAY NOT use integer, boolean, bit string, or duration literals
+	- Why: These are only usable with other language features I am
+      excluding; hence we can exclude these.
+5. MAY NOT use any other compile-time constant expressions
+	- Why: Reduce barrier to entry for implementors. It's easy to
+      imagine a library (or even a visitor in the OpenQASM repo)
+      folding constants, so we'd end up at the literals.
+6. MAY NOT use qubit registers or classical registers
+	- Why: Register manipulations require bounds checking -- why bother.
+7. MAY NOT declare qubits (`qubit <name>;`)
+	- Why: Use of user-defined qubits implies that language processors
+      manage a symbol table, and that implementors support some kind
+      of physical qubit mapping/allocation. The latter typically
+      imposes some constraints and to state clear assumptions here
+      would require going beyond a "minimal" profile.
+8. MAY use physical qubits
+	- Why: Easy for language processors, and more or less
+      unavoidable. Implementors will still have constraints on the
+      number of qubits and their supported operations.
+9. MAY NOT declare new gates
+	- Why: The "gates = unitaries" model of OpenQASM is far from the
+      reality of hardware implementations. This implies some sort of
+      compilation -- we can assume that this has already been done for
+      us.
+10. MAY use  "built-in gates" or gates defined in `stdgates.inc`
+	- Why: Hard to avoid this, but by merit of this being a fixed set
+      of gates implementations can hardcode the implementation as
+      needed.
+11. MAY NOT use gate modifiers
+	- Why: Unnecessary complexity at this level -- we assume that
+      these have been compiled away.
+12. MAY NOT use broadcasting in gate applications
+	- Why: This is implied in our restriction to physical qubits, but
+      worth stating more explicitly: this is unnecessary complexity.
+13. MAY use `reset` as the first operation on a physical qubit
+	- Why: Not sure if the spec assumes zero-initialization otherwise,
+      but this is easy to support or ignore in pretty much all
+      implementations I am aware of.
+14. MAY NOT use `reset` in any other context
+	- Why: I wouldn't want to require implementors support active reset.
+15. MAY use `measure` statements as the last operation on a physical qubit
```

Comment:
It's unclear from this statement whether `measure` can be used as the non-last operation on a qubit. I propose we explicitly state that this is not allowed.
Adding support for this would enable things like non-demolition measurements, but I think this doesn't belong in the minimal set

Reply:
What about:
```MAY use `measure` statements, but only as the last operation on a physical qubit```